### PR TITLE
Add target platform flag to build script

### PR DESCRIPTION
**Pull request by dsimansk**
## Changes

* Add target platform flag to build script

## Reference

Fixes #

**Release Note**

```release-note
Add target `--platform` flag to build script
```

### CODE REVIEW

**Comment by dsimansk** [comment]
@cardil @rhuss PTAL

File: hack/build.sh [hunk 1]
```diff
@@ -146,6 +155,25 @@ go_build() {
...
+
+  if [ -z "${GOOS}" ] || [ -z "${GOARCH}" ]; then
+    echo "❌ Missing GOOS or GOARCH. Please provide both GOOS and GOARCH as arguments."
+    exit 1
+  fi
+
+  echo "🚧 Compile for GOOS=${GOOS} GOARCH=${GOARCH}"
+
+  # Env var exported by hack/build-flags.sh
+  GOOS="${GOOS}" GOARCH="${GOARCH}" go build -mod=vendor -ldflags "${KN_BUILD_LD_FLAGS:-}" -o kn ./cmd/...
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply hack/build.sh]: Thanks for the review, updated!

**Review by cardil — APPROVED** [review]
/lgtm

**Pull request merged**